timesketch/lib/sigma_util_test.py

```diff
@@ -26,7 +26,93 @@
 from timesketch.lib import sigma_util
 
 
-MOCK_SIGMA_RULE = """
+SIGMA_MOCK_RULE_TEST4 = r"""
+title: Login with WMI
+id: 5af54681-df95-4c26-854f-2565e13cfab0
+status: stable
+description: Detection of logins performed with WMI
+author: Thomas Patzke
+date: 2019/12/04
+tags:
+    - attack.execution
+    - attack.t1047
+logsource:
+    product: windows
+    service: security
+detection:
+    selection:
+        EventID: 4624
+        ProcessName|endswith: '\WmiPrvSE.exe'
+    condition: selection
+falsepositives:
+    - Monitoring tools
+    - Legitimate system administration
+level: low
+"""
+
+
+class TestSigmaUtilLib(BaseTest):
+    """Tests for the sigma support library."""
+
+    def test_sanitize_rule_string(self):
+        """Testing the string sanitization"""
+
+        # pylint: disable=protected-access
+        test_1 = sigma_util._sanitize_query("(* lorem * OR * lorema *)")
+        self.assertIsNotNone(test_1)
+
+        self.assertEqual(
+            sigma_util._sanitize_query("test.keyword:foobar"),
+            "test:foobar",
+        )
+        self.assertEqual(
+            sigma_util._sanitize_query("(* foobar *)"),
+            '(" foobar ")',
+        )
+        self.assertEqual(sigma_util._sanitize_query("*foo bar*"), '"foo bar"')
+
+        # test that the function does not break regular queries
+        self.assertEqual(
+            sigma_util._sanitize_query(
+                "*mimikatz* OR *mimikatz.exe* OR *mimilib.dll*"
+            ),
+            "*mimikatz* OR *mimikatz.exe* OR *mimilib.dll*",
+        )
+
+        test_2 = sigma_util._sanitize_query("(*a:b* OR *c::d*)")
+        self.assertEqual(test_2, r'("a:b" OR "c\:\:d")')
+
+        test_3 = sigma_util._sanitize_query(
+            '(xml_string.keyword:"\\foobar.exe" AND GrantedAccess.keyword:"10")'
+        )
+
+        self.assertEqual(
+            test_3, r'(xml_string:"\foobar.exe" AND GrantedAccess:"10")'
+        )
+
+        test_4 = sigma_util._sanitize_query(
+            '(xml_string:C:\\Program Files\\WindowsApps\\\" AND xml_string: "GamingServices.exe)'  # pylint: disable=line-too-long
+        )
+        self.assertIsNotNone(test_4)
+        # pylint: enable=protected-access
```

Comment:
What's the difference between test_2, test_3, test_4? Any chance we can actually split these into different tests, or parametrize them?